search/search-api/api/src/java/org/sakaiproject/search/api/SearchService.java

```diff
@@ -156,6 +158,62 @@ SearchList search(String searchTerms, List<String> contexts, int searchStart,
 	 */
 	public SearchList search(String searchTerms, List<String> contexts, int start,
 			int end, String filterName, String sorterName) throws InvalidSearchQueryException;
+
+	/**
+	 * This is the same as standard search, but the caller can specify the logical name
+	 * of the index builder which should execute the search.
+	 *
+	 *
+	 * @param searchTerms A search string
+	 * @param contexts A list of contexts
+	 * @param start starting from
+	 * @param end ending at
+	 * @param indexBuilderName logical index builder name or {@code null} to fall back to default
+	 * @return
+	 * @throws InvalidSearchQueryException if unable to parse the query
+     */
+	SearchList search(String searchTerms, List<String> contexts, int start,
```

Comment:
Yeah instead of some vague data structure just create a object that contains all data and pass that in. I would suggest something like the builder pattern for creating it.